src/lido/infos.ts

```diff
@@ -0,0 +1,126 @@
+
+    const vaultAccountInfo = await connection.getAccountInfo(vaultId);
+    if (!vaultAccountInfo) throw Error("Error: Cannot get solido token account");
+
+    const vault: types.VaultInfo = this.parseVault(vaultAccountInfo.data, vaultId);
+
+    return vault;
+  }
+
+  static parseVault(data: Buffer, vaultId: PublicKey): types.VaultInfo {
+    // Decode the Token data using AccountLayout
+    const decodeData = LIDO_LAYOUT.decode(data);
+    const {
+      lidoVersion,
+      manager,
+      stSolMint,
+      exchangeRate,
+      solReserveAuthorityBumpSeed,
+      stakeAuthorityBumpSeed,
+      mintAuthorityBumpSeed,
+      rewardsWithdrawAuthorityBumpSeed,
+      rewardDistribution,
+      feeRecipients,
+      metrics,
+      validators,
+      maintainers,
+    } = decodeData;
+
+    // Ensure the mint matches
+    if (!this._isAllowed(stSolMint)) {
+      throw Error("Error: Not a stSOL token account");
+    }
+
+    return {
+      vaultId,
+      lidoVersion,
+      manager,
+      shareMint: stSolMint,
+      exchangeRate,
+      solReserveAuthorityBumpSeed,
+      stakeAuthorityBumpSeed,
+      mintAuthorityBumpSeed,
+      rewardsWithdrawAuthorityBumpSeed,
+      rewardDistribution,
+      feeRecipients,
+      metrics,
+      validators,
+      maintainers,
+    };
+  }
+
+  static async getAllDepositors(connection: Connection, userKey: PublicKey): Promise<types.DepositorInfo[]> {
+    const { value: accountInfos } = await connection.getTokenAccountsByOwner(userKey, {
+      mint: ST_SOL_MINT_ADDRESS,
+    });
+    const depositors = accountInfos.map((depositor) => this.parseDepositor(depositor.account.data, depositor.pubkey));
+    return depositors;
+  }
+
+  static async getDepositor(connection: Connection, depositorId: PublicKey): Promise<types.DepositorInfo> {
+    const depositorAccountInfo = await connection.getAccountInfo(depositorId);
+    if (!depositorAccountInfo) throw Error("Error: Cannot get depositor account");
+
+    return this.parseDepositor(depositorAccountInfo.data, depositorId);
+  }
+
+  static getDepositorId(vaultId: PublicKey, _userKey: PublicKey, _programId: PublicKey = LIDO_PROGRAM_ID): PublicKey {
+    return vaultId;
+  }
+
+  static parseDepositor(data: Buffer, depositorId: PublicKey): types.DepositorInfo {
+    const decodeData = LIDO_TOKEN_LAYOUT.decode(data);
+    const { mint, amount, owner } = decodeData;
+
+    // Ensure the mint matches
+    if (!this._isAllowed(mint)) {
+      throw Error(`Error: Not a stSOL token account`);
+    }
+
+    return {
+      depositorId,
+      userKey: owner,
+      amount,
+      mint,
+    };
+  }
+
+  private static _isAllowed(mint: PublicKey): boolean {
+    return mint.equals(ST_SOL_MINT_ADDRESS);
+  }
+};
+
+export { infos };
+
+export class VaultInfoWrapper implements IVaultInfoWrapper {
+  constructor(public vaultInfo: types.VaultInfo) {}
+
+  getApr() {
+    // TODO
+    return 0;
+  }
```

Comment:
Try to fetch the ARY in `getAllVaults` and `getVault` from their API, and add an additional attribute `apr` in `VaultInfo`. 
Return `5.74`(default vault set in Lido SDK) if `this.vualtInfo.apr` is `undefined` in `getApr()`.
Can check more details [here](https://github.com/lidofinance/solido-sdk/blob/main/src/api/stakeApy.ts#L11)

Reply:
Hey, since this function will be in the `VaultInfoWrapper`, could it be better to have all the logic there?

I mean fetch it inside `VaultInfoWrapper` using the API, which I think might be better because the `apr` field could be `undefined` in `VaultInfo`, so I'm guessing the primary method of interacting with it will be through the wrapper.

Reply:
Also, could you help me out with the differences in APR and APY here? I'm a little confused between them.

Reply:
> Hey, since this function will be in the VaultInfoWrapper, could it be better to have all the logic there?

Actually, we would prefer done all fetching stuff at initialization stage. But like you mention above, `apr` field might sometimes be misused and might result in `undefined`, so this idea lgtm.

> could you help me out with the differences in APR and APY here?

it would be better using `getApy` instead of `getApr` here, since the staked SOL will compound.